hyp3_testing/helpers.py

```diff
@@ -73,3 +76,37 @@ def clarify_xr_message(message: str, left: str = 'reference', right: str = 'seco
     message = message.replace('\nL ', f'\n\n{left[0].upper()} ')
     message = message.replace('\n\n\n', '\n\n')
     return message
+
+
+def determine_product_files(job_instance):
+    product_archive = job_instance.files[0]['url']
+
+    with RemoteZip(product_archive) as z:
+        files = z.infolist()
+
+    product_name = files[0].filename.rstrip('/')
+
+    hash_name = product_name.split('_')[-1]
+    files_normalized = {f.filename.replace(hash_name, 'HASH') for f in files if not f.is_dir()}
+
+    return product_name, files_normalized
+
+
+@contextmanager
+def job_tifs(job_id, api, directory, keep=False):
+    hyp3 = HyP3(api, os.environ.get('EARTHDATA_LOGIN_USER'), os.environ.get('EARTHDATA_LOGIN_PASSWORD'))
+    job = hyp3.get_job_by_id(job_id)
+
+    product_dir = directory / job.files[0]['filename'].replace('.zip', '')
+    if not product_dir.exists():
```

Comment:
This *might* not be robust enough as it's just based on the directory. We could also add a `--force-download` option or something which ensure we re-download everything.